### ISSUE TITLE
CollisionTree bug

### ISSUE THREAD

**Issue by runiter** [gouessej/Ardor3D#23]
There is a rare but fatal bug in CollisionTree. If mesh initially has 16 or more primitives, then its primitives reduced to 15 or less, it causes exception when picking:

```
java.lang.IndexOutOfBoundsException: Invalid primitiveIndex '15'.  Count is 15
	at com.ardor3d.scenegraph.MeshData.getPrimitiveVertices(MeshData.java:822)
	at com.ardor3d.bounding.CollisionTree.intersect(CollisionTree.java:476)
	at com.ardor3d.bounding.CollisionTree.intersect(CollisionTree.java:466)
	at com.ardor3d.scenegraph.Mesh.intersectsPrimitivesWhere(Mesh.java:568)
	at com.ardor3d.intersection.PrimitivePickData.<init>(PrimitivePickData.java:22)
	at com.ardor3d.intersection.PrimitivePickResults.addPick(PrimitivePickResults.java:29)
	at com.ardor3d.intersection.PickingUtil.findPick(PickingUtil.java:59)
```

It's due to forgetting to clear the left and right children:

https://github.com/gouessej/Ardor3D/blob/master/ardor3d-core/src/main/java/com/ardor3d/bounding/CollisionTree.java#L227

        if (_end - _start + 1 <= CollisionTreeManager.getInstance().getMaxPrimitivesPerLeaf()) {
            return;
        }

I think the solution is to do this:

        if (_end - _start + 1 <= CollisionTreeManager.getInstance().getMaxPrimitivesPerLeaf()) {
            _left = null;
            _right = null;
            return;
        }

But I couldn't test it since I'm using the Ardor3D jar files.

**Comment by gouessej** [comment]
Good findings! Thank you. I will commit your fix and rebuild Ardor3D Tuesday or Wednesday. Sorry for the delay.

**Comment by runiter** [comment]
No worries, I'm in no rush because I already addressed the problem by simply replacing all calls to CollisionTreeManager.INSTANCE.updateCollisionTree() with CollisionTreeManager.INSTANCE.removeCollisionTree() in our software. In our case this solution is already sufficient specially because we won't be able to update the Ardor3D jar files any time soon. I only reported this problem to be helpful to others. In future I will try to send a pull request.
